### PR TITLE
"add go compile option"

### PR DESCRIPTION
**Pull request by dzhwinter**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -126,9 +127,12 @@ endif(WITH_GPU)
 
 add_subdirectory(proto)
 add_subdirectory(paddle)
-add_subdirectory(go/master/c)
 add_subdirectory(python)
+
+if(WITH_GO)
+add_subdirectory(go/master/c)
 add_subdirectory(go/pserver/cclient)
```
Comment (jacquesqiao) [inline CMakeLists.txt]: 这部分可以先挪出来

**Comment by jacquesqiao** [comment]
https://github.com/PaddlePaddle/Paddle/blob/develop/python/CMakeLists.txt#L21

**Comment by dzhwinter** [comment]
done.

File: python/CMakeLists.txt [hunk 2]
```diff
@@ -15,10 +15,12 @@ set(PY_FILES paddle/__init__.py
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in
     ${CMAKE_CURRENT_BINARY_DIR}/setup.py)
 
-add_custom_command(OUTPUT ${OUTPUT_DIR}/.timestamp
-    COMMAND env ${py_env} ${PYTHON_EXECUTABLE} setup.py bdist_wheel
-    COMMAND ${CMAKE_COMMAND} -E touch ${OUTPUT_DIR}/.timestamp
-    DEPENDS gen_proto_py ${PY_FILES} ${external_project_dependencies} paddle_master_shared)
+if(WITH_Go)
```
Comment (jacquesqiao) [inline python/CMakeLists.txt]: 这个地方只需要去掉修改
```shell
DEPENDS gen_proto_py ${PY_FILES} ${external_project_dependencies} paddle_master_shared)
```
去掉paddle_master_shared的依赖

**Comment by dzhwinter** [comment]
we have solved this problem it current release.

**Pull request closed without merging**